src/nanotron/trainer.py

```diff
@@ -654,13 +665,17 @@ def _load_model_checkpoint(self, model: NanotronModel) -> NanotronModel:
                     parallel_context=self.parallel_context,
                     root_folder=self.config.model.init_method.path,
                 )
-            elif isinstance(self.config.model.init_method, RandomInit):
-
-                unwrapped_model.init_model_randomly(config=self.config)
+            elif isinstance(self.config.model.init_method, (RandomInit, SpectralMupInit)):
+                _init_method = (
+                    ParametrizationMethod.STANDARD
+                    if isinstance(self.config.model.init_method, RandomInit)
+                    else ParametrizationMethod.SPECTRAL_MUP
+                )
+                unwrapped_model.init_model_randomly(config=self.config, init_method=_init_method)
```

Comment:
why not keep only config in `init_model_randomly` and define the init_method inside the function using `config` (that makes for less code in trainer)